src/sage/schemes/elliptic_curves/jacobian.py

```diff
@@ -96,6 +96,16 @@ def Jacobian(X, **kwds):
                 (-u^4*v^4*w - u^4*v*w^4 - u*v^4*w^4 :
                  1/2*u^6*v^3 - 1/2*u^3*v^6 - 1/2*u^6*w^3 + 1/2*v^6*w^3 + 1/2*u^3*w^6 - 1/2*v^3*w^6 :
                  u^3*v^3*w^3)
+
+    TESTS:
+
+    Check that the following doesn't segmentation fault
+    (the error message could be improved)::
+
+        sage: Jacobian(GF(11)['x,y'](3))
+        Traceback (most recent call last):
+        ...
+        AssertionError
     """
```

Comment:
The docstring says
```
* "X" -- polynomial, algebraic variety, or anything else that has a
  Jacobian elliptic curve
```
Is `GF(11)['x,y'](3)` a proper input? 

Preventing segmentation fault even for improper inputs seems to be a good thing. But the "TEST" seems to be misplaced. A "TEST" may be added to `src/sage/geometry/integral_points.pxi`. Or, in my opinion, no "TEST" is acceptable since it is unlikely to break again.

Reply:
I believe it's not a proper input. the polynomial `3` defines an empty scheme, if I recalled correctly that's not a variety.

not sure whether the test is misplaced. It is testing the Jacobian function, it just so happen that that calls this function.

Maybe a separate test that tests directly this function with invalid input is in order. I'll need to figure out.